doc/source/how-to-use-driver-api.rst

```diff
@@ -0,0 +1,254 @@
+
+.. admonition:: Note
+    :class: note
+
+    Each :code:`Message` contains a |recordset_link|_ object. It contains :code:`parameters_records`, :code:`metrics_records`, and :code:`configs_records` attributes, which are - unsurprisingly - parameters, metrics, and configurations that are used by a node to execute a task. 
+
+Here is an example of how to push a PyTorch model and instructions to a set of client node IDs to train the model. First, we define a utility function to convert a PyTorch model into a :code:`ParametersRecord` and create a PyTorch model in :code:`main()` (we’ve omitted the implementation details for :code:`Net()`, but you can refer to :doc:`this quickstart tutorial <tutorial-quickstart-pytorch>` for an example):
+
+.. code-block:: python
+
+    # File: server.py
+
+    def pytorch_to_parameter_record(pytorch_module: torch.nn.Module):
+        state_dict = pytorch_module.state_dict()
+        for k, v in state_dict.items():
+            state_dict[k] = _ndarray_to_array(v.numpy())
+        return ParametersRecord(state_dict)
+
+
+    @app.main()
+    def main(driver: Driver, context: Context) -> None:
+    	global_model = Net()
+
+Then, we create a :code:`RecordSet` and add parameters and configurations to :code:`parameters_record` and :code:`configs_record`, respectively. Note that the dictionary keys used here are customizable, so you have a great degree of flexibility to assign and use the dictionaries in a :code:`RecordSet`:
+
+.. code-block:: python
+
+    # File: server.py
+    # In the for-loop in `main()`
+
+    # Create a RecordSet
+    recordset = RecordSet()
+
+    # Add model parameters to the RecordSet
+    recordset.parameters_records["my_model"] = pytorch_to_parameter_record(global_model)
+
+    # Add a training configuration for 1 epoch only to the RecordSet
+    recordset.configs_records["my_config"] = ConfigsRecord({"epochs": 1})
+
+Next, we create a list of :code:`Messages`, one for each node ID. To do so, we loop over all node IDs and run :code:`Driver.create_message()` with the :code:`recordset` as the content of the message: 
+
+.. code-block:: python
+
+    # File: server.py
+
+    messages = []
+    for node_id in node_ids:
+        message = driver.create_message(
+            content=recordset,
+            message_type=MessageType.TRAIN,
+            dst_node_id=node_id,
+            group_id=str(server_round),
+            ttl=DEFAULT_TTL,
+        )
+        messages.append(message)
+
+Finally, we use :code:`Driver.push_messages()` to push the list of :code:`Messages` containing the encapsulated parameters and configurations to the nodes.
+
+.. code-block:: python
+
+    # File: server.py
+
+    message_ids = driver.push_messages(messages)
+
+:code:`Driver.push_messages()` yields an iterable list of message IDs. In some real-world scenarios, you may encounter situations where only some :code:`Messages` can be pushed, so it is good practice to filter out empty message IDs:
```

Comment:
I think we should ellaborate on these "real-world" scenarios. For now, I'm only aware that a returned message_id will be empty if the `Message` is not well-formed (i.e. there are some inconsistencies in its `Metadata`). What are the other scenarios where `push_messages(0)` can return `""`, @panh99 ?

Reply:
@jafermarq @panh99 How about for now, I emphasize that this is a "good practice" step. Later we can open another PR to elaborate on the scenarios if required? I can reword the sentence to this: 

"In some rare scenarios, you may encounter situations where not all `Message` objects can be pushed, e.g. if a `Message` is not well formed, so it is good practice to filter out empty message IDs:"

What do you think?